app/models/concerns/audit_extensions.rb

```diff
@@ -33,8 +33,19 @@ def ensure_taxonomies_not_escalated
   private
 
   def log_audit
-    Foreman::Logging.with_fields(self.audited_changes) do
-      Foreman::Logging.logger('audit').info { "#{self.action} event for #{self.auditable_type} with id #{self.auditable_id}" }
+    return unless Foreman::Logging.logger('audit').info?
+    self.audited_changes.each_pair do |column_name, old_and_new|
+      audited_fields = {
+        audit_action: self.action,
+        audit_type: self.auditable_type,
+        audit_id: self.auditable_id,
+        audit_column: column_name,
+        audit_old: old_and_new ? old_and_new[0] : '',
+        audit_new: old_and_new ? old_and_new[1] : '',
+      }
+      Foreman::Logging.with_fields(audited_fields) do
+        Foreman::Logging.logger('audit').info { "#{self.auditable_type} (#{self.auditable_id}) #{self.action} event on #{column_name}" }
```

Comment:
Passing argument as block is no longer needed as you added a guard condition on top, just pass the string

Reply:
Removed.